File: src/ipc/commands/summary.ts
import { invokeCommand } from "../client";
import type { Summary } from "../types";

export function summaryGenerate(meetingId: string): Promise<Summary> {
  return invokeCommand<Summary>("summary_generate", {
    meeting_id: meetingId,
  });
}

export function summaryGet(meetingId: string): Promise<Summary> {
  return invokeCommand<Summary>("summary_get", { meeting_id: meetingId });
}

File: src-tauri/src/commands/summary.rs
use tauri::State;

use crate::error::CmdResult;
use crate::models::Summary;
use crate::services;
use crate::AppState;

/// `rename_all = "snake_case"` keeps IPC args aligned with api-shape.md / `src/ipc`.
#[tauri::command(rename_all = "snake_case")]
pub fn summary_generate(
    state: State<'_, AppState>,
    meeting_id: String,
) -> CmdResult<Summary> {
    let conn = state
        .db
        .lock()
        .map_err(|_| crate::error::AppErrorDto::internal("Database lock poisoned"))?;
    services::generate_summary_http(&conn, &meeting_id)
}

#[tauri::command(rename_all = "snake_case")]
pub fn summary_get(state: State<'_, AppState>, meeting_id: String) -> CmdResult<Summary> {
    let conn = state
        .db
        .lock()
        .map_err(|_| crate::error::AppErrorDto::internal("Database lock poisoned"))?;
    services::get_summary(&conn, &meeting_id)
}
